instrumentation/net/http/otelhttp/transport.go

```diff
@@ -109,19 +142,66 @@ func (t *Transport) RoundTrip(r *http.Request) (*http.Response, error) {
 		ctx = httptrace.WithClientTrace(ctx, t.clientTrace(ctx))
 	}
 
+	readRecordFunc := func(int64) {}
+	if t.readEvent {
+		readRecordFunc = func(n int64) {
+			span.AddEvent("read", trace.WithAttributes(ReadBytesKey.Int64(n)))
+		}
+	}
+
+	var bw bodyWrapper
+	// if request body is nil or NoBody, we don't want to mutate the body as it
+	// will affect the identity of it in an unforeseeable way because we assert
+	// ReadCloser fulfills a certain interface, and it is indeed nil or NoBody.
+	if r.Body != nil && r.Body != http.NoBody {
+		bw.ReadCloser = r.Body
+		bw.record = readRecordFunc
+		r.Body = &bw
+	}
+
 	r = r.Clone(ctx) // According to RoundTripper spec, we shouldn't modify the origin request.
 	span.SetAttributes(semconvutil.HTTPClientRequest(r)...)
+	if t.getRequestAttributes != nil {
+		span.SetAttributes(t.getRequestAttributes(r)...)
+	}
 	t.propagators.Inject(ctx, propagation.HeaderCarrier(r.Header))
 
 	res, err := t.rt.RoundTrip(r)
 	if err != nil {
 		span.RecordError(err)
 		span.SetStatus(codes.Error, err.Error())
 		span.End()
-		return res, err
+	} else {
+		span.SetAttributes(httpconv.ClientResponse(res)...)
+		if t.getResponseAttributes != nil {
+			span.SetAttributes(t.getResponseAttributes(res)...)
+		}
+		span.SetStatus(httpconv.ClientStatus(res.StatusCode))
+		res.Body = newWrappedBody(span, res.Body)
+	}
+
+	// Add metrics
+	attributes := httpconv.ClientRequest(r)
+	if t.getRequestAttributes != nil {
+		attributes = append(attributes, t.getRequestAttributes(r)...)
 	}
+	if err == nil {
+		attributes = append(attributes, httpconv.ClientResponse(res)...)
+		if t.getResponseAttributes != nil {
+			attributes = append(attributes, t.getResponseAttributes(res)...)
+		}
+	}
+	o := metric.WithAttributes(attributes...)
+	t.counters[ClientRequestContentLength].Add(ctx, bw.read.Load(), o)
```

Comment:
`bodyWrapper.read` is incremented by 1 on every `Read()` call. So `bw.read.Load()` will be number of reads and not the request content length? or am i missing something?

Reply:
Fixed the couting code.